tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -2016,6 +2021,241 @@ tensorflow::Status ConvertSqueeze(OpConverterParams* params) {
   return tensorflow::Status::OK();
 }
 
+// Gets the bounds (start or end) from the weights of a StridedSlice op.
+tensorflow::Status GetStridedSliceBound(const std::vector<int>& input_dims,
+                                        const TRT_ShapedWeights& bound_weights,
+                                        string bound_name, string node_name,
+                                        std::vector<int>& output_bound) {
+  const int* weights_ptr =
+      static_cast<int*>(const_cast<void*>(bound_weights.GetValues()));
+  output_bound = 
+      std::vector<int>(weights_ptr, weights_ptr + bound_weights.count());
+  if (output_bound.size() != input_dims.size()) {
+    return tensorflow::errors::InvalidArgument(
+        "StridedSlice \"", bound_name, "\" specified ",
+        std::to_string(output_bound.size()), " dimensions, but input rank is ",
+        std::to_string(input_dims.size()), ", at ", node_name);
+  }
+  for (int i = 0; i < output_bound.size(); i++) {
+    // Make sure bound is valid.
+    if ((output_bound[i] < -input_dims[i]) ||
+        (output_bound[i] > input_dims[i])) {
+      return tensorflow::errors::InvalidArgument(
+          bound_name,
+          " for StridedSlice is invalid, must be in the range "
+          "[-rank(input), rank(input)], at ",
```

Comment:
I don't understand this error message with range based on the rank. shouldn't it depend on the input dim?